### PR TITLE
Fix MaxPods test to work with image puller

### PR DESCRIPTION
**Pull request by gmarek**
@yujuhong @bprashanth - please ping me when SchedulerPredicates may be impacted.

@david-mcmahon - This PR fixes MaxPods in serial suite that was broken by introduction of image puller.

cc @davidopp

### CODE REVIEW

File: test/e2e/scheduler_predicates.go [hunk 1]
```diff
@@ -132,6 +132,14 @@ func waitForStableCluster(c *client.Client) int {
 
 	allPods, err := c.Pods(api.NamespaceAll).List(api.ListOptions{})
 	framework.ExpectNoError(err)
+	// API server returns also Pods that succeeded. We need to filter them out.
+	currentPods := make([]api.Pod, 0, len(allPods.Items))
+	for _, pod := range allPods.Items {
+		if pod.Status.Phase != api.PodSucceeded {
```
Comment (wojtek-t) [inline test/e2e/scheduler_predicates.go]: Probably you also want to filter out Failed pods too

Comment (gmarek) [inline test/e2e/scheduler_predicates.go]: Done.

**Comment by wojtek-t** [comment]
LGTM

**Comment by wojtek-t** [comment]
Merging manually to fix tests

**Comment by bprashanth** [comment]
Thanks for fixing this!

> please ping me when SchedulerPredicates may be impacted.

No idea when this may or may not be affected when changes are made outside scheduler code. I can ping you on evey change I make to e2e.go, or we can make the test break merge queue and everyone will get pinged and we'll fix it asap.

**Comment by yujuhong** [comment]
> @yujuhong @bprashanth - please ping me when SchedulerPredicates may be impacted.

Could you describe what will break SchedulerPredicates? If I knew this would've been impacted, I'd have probably fixed it already :-)
There were discussions to make serial suite blocking the merge queue, which will hopefully solve the problem.

**Comment by wojtek-t** [comment]
Yeah - we should make the serial suite blocking.
@davidopp ^^

**Comment by gmarek** [comment]
@yujuhong - anything that changes in any way semantics of what 'Ready' means and anything that introduces expectation that system pods would be in other phase than 'Ready' (and Succeded/Failed after this test). In this particular case running the test after introducing the changes to its code would show that there's an issue.

**Comment by yujuhong** [comment]
@gmarek Sorry for breaking the test (since I reviewed the original PR). 

On the other hand, I can also argue that a scheduler test should not care about terminated pods at all. The reality is that no one can keep track of all the tests (and their specific implementations even if we try), and that's why we have blocking suites.

**Comment by gmarek** [comment]
Yeah - you're right, they should not, and now they don't:). As always in such cases, we learned it a hard way:)

**Comment by wojtek-t** [comment]
I still think that the solution here is not doing anything super sophisticated with tests or to educate everyone about everything. We simply need more blocking suites...

**Pull request merged**